lib/src/runner/reporter/xunit.dart

```diff
@@ -0,0 +1,353 @@
+// Copyright (c) 2015, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+library test.runner.reporter.xunit;
+
+import 'dart:async';
+
+import '../../backend/live_test.dart';
+import '../../backend/state.dart';
+import '../../backend/group.dart';
+import '../../frontend/expect.dart';
+import '../../utils.dart';
+import '../engine.dart';
+import '../load_suite.dart';
+import '../reporter.dart';
+
+class XunitFailureResult {
+  String name;
+  String stack;
+
+  XunitFailureResult(this.name, this.stack);
+}
+
+class XunitTestResult {
+  int beginningTime;
+  int endTime;
+  Map error = {};
+  String name;
+  String path;
+  bool skipped = false;
+  String skipReason;
+
+  XunitTestResult(this.name, this.path, this.beginningTime);
+}
+
+class XunitTestSuite {
+  int errored = 0;
+  int failed = 0;
+  int skipped = 0;
+  List testResults = [];
+  int tests = 0;
+
+  add(XunitTestResult test) {
+    testResults.add(test);
+  }
+}
+
+/// A reporter that returns xunit compatible output.
+
+class XunitReporter implements Reporter {
+  /// The engine used to run the tests.
+  final Engine _engine;
+
+  /// A stopwatch that tracks the duration of the full run.
+  final _stopwatch = new Stopwatch();
+
+  /// Whether we've started [_stopwatch].
+  ///
+  /// We can't just use `_stopwatch.isRunning` because the stopwatch is stopped
+  /// when the reporter is paused.
+  var _stopwatchStarted = false;
+
+  /// Whether the reporter is paused.
+  var _paused = false;
+
+  /// The set of all subscriptions to various streams.
+  final _subscriptions = new Set<StreamSubscription>();
+
+  /// Record of all tests run.
+  Map _testCases = {};
+
+  /// Number of failures that occur during the test run.
+  int _failureCount = 0;
+
+  /// Number of errors that occur during the test run.
+  int _errorCount = 0;
+
+  /// Map containing Xunit hierarchy.
+  Map _groupStructure = {};
+
+  /// Watches the tests run by [engine] and records its results.
+  static XunitReporter watch(Engine engine, {bool verboseTrace: false}) {
```

Comment:
i thought it would be necessary when I was initially starting, but I removed all references to it directly in the code but not from here